t/08-autotest.t

```diff
@@ -86,6 +86,35 @@ is($died,      0, 'start+next+start should not die');
 is($completed, 1, 'start+next+start should complete');
 @sent = [];
 
+# Test loading snapshots with force_reset flag. Have to put it here, before loading
+# runargs test module, as it fails.
+# Test that snapshot loading is triggered even when tests are successful
+$mock_basetest->mock(test_flags => sub { return { force_reset => 1 }; });
+$mock_autotest->mock(query_isotovideo => sub { return 1; });
+my $reverts_done = 0;
+$mock_autotest->mock(load_snapshot => sub { $reverts_done++; });
+
+autotest::run_all;
+($died, $completed) = get_tests_done;
+is($died,      0, 'start+next+start should not die when force_reset flag is set');
+is($completed, 1, 'start+next+start should complete when force_reset flag is set');
+is($reverts_done, 0, "No snapshots loaded if not test with milestone flag");
+$reverts_done = 0;
+@sent = [];
+# Test with snapshot available
+$mock_basetest->mock(test_flags => sub { return { force_reset => 1, milestone => 1 }; });
+autotest::run_all;
+($died, $completed) = get_tests_done;
+is($died,      0, 'start+next+start should not die when force_reset flag is set');
+is($completed, 1, 'start+next+start should complete when force_reset flag is set');
+is($reverts_done, 2, "Snapshots are loaded even when tests succeed");
+@sent = [];
```

Comment:
hm, do we have a test now that tests that *without* `force_reset` *no* revert is done on success?